src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import java.io.IOException;
+import java.time.LocalDate;
+
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructor for Duke class
+     *
+     * @param filePath
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (IOException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Method to run entire programme flow
+     */
+    public void run() {
```

Comment:
This method is quite long. Maybe you can extract actions for each command as separate methods :)